docs/DeveloperGuide.md

```diff
@@ -1855,42 +1855,696 @@ testers are expected to do more _exploratory_ testing.
 
-### Deleting a customer
+2. Subsequent launches
 
-1. Deleting a customer while all customers are being shown
+   1. Relaunch the application by running `HomeBoss.jar`{.swift}.
 
-    1. Prerequisites: List all customers using the `list` command. Multiple customers in the list.
+   2. Using the `login`{.swift} command, log in into HomeBoss with the same user details entered earlier.
+      Expected: User is able to log in successfully and see the HomeBoss homepage.
 
-    1. Test case: `delete 1`<br>
-       Expected: First contact is deleted from the list. Details of the deleted contact shown in the status message.
-       Timestamp in the status bar is updated.
+### Register
 
-    1. Test case: `delete 0`<br>
-       Expected: No customer is deleted. Error details shown in the status message. Status bar remains the same.
+1. Registering for an account
 
-    1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
-       Expected: Similar to previous.
+   1. Prerequisites: None
 
-1. _{ more test cases …​ }_
+   2. Test Case: `register --user Gabriel --password GabrielIsGreat --confirmPass GabrielIsGreat --secretQn First Pet Name? --answer Koko`{.swift}.</br>
+      Expected: A new user account is registered with username `Gabriel`{.swift}, password `GabrielIsGreat`{.swift},
+      secret question `First Pet Name?`{.swift} and answer `Koko`{.swift}. 
+
+   3. Test Case: `register --user Ga_briel --password GabrielIsGreat --confirmPass GabrielIsGreat --secretQn First Pet Name? --answer Koko`{.swift}.</br>
+      Expected: No new user is registered. Error indicating username constraints is
+      shown in the feedback message.
+
+   4. Test Case: `register --user Gabriel --password Gabriel_IsGreat --confirmPass Gabriel_IsGreat --secretQn First Pet Name? --answer Koko`{.swift}.</br>
+      Expected: No new user is registered. Error indicating password constraints is
+      shown in the feedback message.
+
+   5. Test Case: `register --user Gabriel --password GabrielIsGreat --confirmPass GabrielIsGreat1 --secretQn First Pet Name? --answer Koko`{.swift}.</br>
+      Expected: No new user is registered. Error indicating password and confirm password do not match is
+      shown in the feedback message.
+
+   6. Test Case: `register --user Gabriel --password GabrielIsGreat --confirmPass GabrielIsGreat --secretQn First Pet Name?`{.swift}.</br>
+      Expected: No new user is registered. Error indicating invalid command format is
+      shown in the feedback message.
+
+   7. Test Case: There already exists a user account stored in the application.</br>
+      Expected: No new user is registered. Error indicating existing account is
+      shown in the feedback message.
+
+### Login
+
+1. Login to an account
+
+   1. Prerequisites: There exists a stored user in the application, with the username `Gabriel`{.swift} and
+      password `GabrielIsGreat`{.swift}
```

Comment:
Perhaps can add in the condition that the user should be logged out as it causes different outcome for the first test case.